### ISSUE TITLE
GRIZZLY0013: Exception during FilterChain execution

### ISSUE THREAD

**Issue by DaveJarvis** [javaee/glassfish#22407]
Install [JDK 8u161](http://www.oracle.com/technetwork/java/javase/downloads/jdk8-downloads-2133151.html) such that upgrades can be made by changing a symbolic link as follows:

    cd /opt
    tar xf $HOME/jdk*gz
    ln -s jdk1* jdk

Allow all users to run the JDK, system wide:

1. Edit `/etc/environment`
1. Append `JAVA_HOME=/opt/jdk`
1. Set `PATH` to: `PATH=/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/opt/jdk/bin`
1. Save the file.
1. Log out and log back in (or `source /etc/environment`).

Unzip GlassFish 5 into `/opt/glassfish5`, then:

    cd /opt/glassfish5/bin
    ./asadmin create-domain fitness
    ./asadmin start-domain fitness
    ./asadmin --host hostname.com --port 4848 enable-secure-admin
    ./asadmin restart-domain fitness

# Expected Results

The application starts.

# Actual Results

Console shows:

    NCLS-ADMIN-00010

Log shows:

``` .log
[2018-01-20T21:41:57.834-0500] [glassfish 5.0] [WARNING] [] [org.glassfish.grizzly.filterchain.DefaultFilterChain] [tid: _ThreadID=43 _ThreadName=admin-listener(3)] [timeMillis: 1516502517834] [levelValue: 900] [[
  GRIZZLY0013: Exception during FilterChain execution
java.lang.NoClassDefFoundError: sun/security/ssl/SupportedEllipticCurvesExtension
        at sun.security.ssl.HelloExtensions.<init>(HelloExtensions.java:82)
        at sun.security.ssl.HandshakeMessage$ClientHello.<init>(HandshakeMessage.java:362)
        at sun.security.ssl.ServerHandshaker.processMessage(ServerHandshaker.java:223)
        at sun.security.ssl.Handshaker.processLoop(Handshaker.java:984)
        at sun.security.ssl.Handshaker$1.run(Handshaker.java:924)
        at sun.security.ssl.Handshaker$1.run(Handshaker.java:921)
        at java.security.AccessController.doPrivileged(Native Method)
        at sun.security.ssl.Handshaker$DelegatedTask.run(Handshaker.java:1379)
        at org.glassfish.grizzly.ssl.SSLUtils.executeDelegatedTask(SSLUtils.java:274)
        at org.glassfish.grizzly.ssl.SSLBaseFilter.doHandshakeStep(SSLBaseFilter.java:708)
```

Visiting http://hostname.com:4848/ fails with **Secure connection failed**.

**Comment by arindam-bandyopadhyay** [comment]
Last know openjdk version where this issue is NOT present is openjdk version "1.8.0_151"

**Comment by yaminikb** [comment]
It worked for me with 1.8.0_152 as well. Looks like Grizzly has some issue loading with u161/u162

**Comment by yaminikb** [comment]
Here is the Grizzly issue for this problem:
https://github.com/javaee/grizzly-npn/issues/4

**Comment by yaminikb** [comment]
The issue is fixed in via PR #22448 

I have verified that secure admin works.

**Comment by doukson2002** [comment]
@yaminikb  will the version located http://download.oracle.com/glassfish/5.0.1/nightly/latest-glassfish.zip
have the grizzly fix?

**Comment by yaminikb** [comment]
@doukson2002 Yes, latest GlassFish has the Grizzly fix.

**Comment by sandeepgholve** [comment]
Worked for me too... :)

**Comment by doukson2002** [comment]
Thank you @Yamini

**Comment by yooouuri** [comment]
Does this also fix #22436?

**Comment by skorejo** [comment]
I am facing same issue with Glassfish 5 and jdk 8_161, I tried jdk 8_151 but there were some weld issues coming , tried jdk8_152 but it still does'nt working with it.
